store/iavl/store.go

```diff
@@ -29,13 +31,19 @@ var (
 
 // Store Implements types.KVStore and CommitKVStore.
 type Store struct {
-	tree Tree
+	tree       Tree
+	pruning    types.PruningOptions
+	version    int64
```

Comment:
The `lastCommit` contains the version as well. Why maintain two version fields? Would they ever differ?

Reply:
Yes the `lastCommit` contains the version number that was last persisted to disk, the store.version will reflect the version we are currently on (even if this version is not flushed to disk).

So if we are on block 60 and flush every 50 blocks. `store.lastCommit.Version() = 50` and `store.Version() = 60`

 we need the commitStore to tell us the version we are currently on for the ABCI to work correctly.

Ref links:
https://github.com/cosmos/cosmos-sdk/blob/aditya/pruning-fixes/baseapp/baseapp.go#L442

https://github.com/cosmos/cosmos-sdk/pull/5579/files#diff-1a66280186fb23f75ab1d32920d81a1fR302

Reply:
Technically, the only place this is absolutely necessary as far as i can tell is in the `CommitMultiStore`. But I added this method to the `CommitStore` interface, hence the Version methods on other stores.

This still seems like the cleaner approach to me

Reply:
Got it. Maybe add some documentation to either the `Store` type or to the individual fields themselves?